pennylane/qnode.py

```diff
@@ -465,6 +467,16 @@ def interface(self, value):
         self._interface = INTERFACE_MAP[value]
         self._update_gradient_fn()
 
+    @property
+    def transform_program(self):
+        """The transform program used by the QNode"""
+        return self._transform_program
+
+    def add_transform(self, transform_container):
+        """Add a transform container to the transform program."""
+        # Temporary append until TransformProgram is added.
+        self._transform_program.append(transform_container)
+

```

Comment:
For now, we could simply do `qnode.transform_program.append(transform_container)`.  That way we can avoid putting even more responsibilities on the `QNode`. The transform program knows how to add things to it, but the QNode doesn't need to care about what the transform program looks like.

Once all the transforms behaviour is all, we can go back and reassess whether or not we need this method. Given how much harder it is to remove behavior than add it, I tend to be a bit more conservative in adding public methods to key classes if there isn't another way.